app/controllers/api/irs_attempts_api_controller.rb

```diff
@@ -18,17 +18,37 @@ class IrsAttemptsApiController < ApplicationController
     respond_to :json
 
     def create
-      if timestamp
-        result = encrypted_security_event_log_result
-
-        headers['X-Payload-Key'] = Base64.strict_encode64(result.encrypted_key)
-        headers['X-Payload-IV'] = Base64.strict_encode64(result.iv)
-
-        send_data result.encrypted_data,
+      if timestamp 
+        
+        if IdentityConfig.store.irs_attempt_api_aws_s3_enabled 
+          if (IrsAttemptApiLogFile.find_by(requested_time: timestamp))
```

Comment:
one more quick note (can be for a follow-up PR), I think that we need to migrate `requested_time` to be a string column in the DB, see previous comment: https://github.com/18F/identity-idp/pull/7323#discussion_r1019664525

Reply:
This has been incorporated into this PR, please comment if you'd like to see it done differently, @zachmargolis!

Reply:
the column change looks good! now I'd update the code that reads to + queries the column to make sure they are formatted the same way (maybe use the same timestamp format as from the filename)?